scrapi/harvesters/usgs.py

```diff
@@ -0,0 +1,124 @@
+from __future__ import unicode_literals
+
+import json
+import logging
+from datetime import date, timedelta
+
+import six
+from nameparser import HumanName
+from scrapi import requests, settings
+from scrapi.base import JSONHarvester
+from scrapi.linter.document import RawDocument
+from scrapi.base.helpers import default_name_parser, compose, build_properties, datetime_formatter
+
+logger = logging.getLogger(__name__)
+
+
+def process_contributors(authors):
+    all_processed_authors = []
+
+    for author in authors:
+        author_d = {}
+        author_d['name'] = author['text']
+        name = HumanName(
+            author['text']
+        )
+        author_d['additionalName'] = name.middle
+        author_d['givenName'] = name.first
+        author_d['familyName'] = name.last
+    return all_processed_authors
+
+
+class USGSHarvester(JSONHarvester):
+    short_name = 'usgs'
+    long_name = 'United States Geological Survey'
+    url = 'https://pubs.er.usgs.gov/'
+    DEFAULT_ENCODING = 'UTF-8'
+
+    URL = 'https://pubs.er.usgs.gov/pubs-services/publication?'
+
+    schema = {
+        'title': '/title',
+        'description': '/docAbstract',
+        'providerUpdatedDateTime': ('/lastModifiedDate', datetime_formatter),
+        'uris': {
+            'canonicalUri': ('/id', 'https://pubs.er.usgs.gov/publication/{}'.format),
+            'providerUris': [('/id', 'https://pubs.er.usgs.gov/publication/{}'.format)],
+            'descriptorUris': [('/doi', 'https://dx.doi.org/{}'.format)]
+        },
+
+        'contributors': ('/contributors/authors', process_contributors),
+        'otherProperties': build_properties(
+            ('serviceID', ('/id', str)),
+            ('definedType', '/defined_type'),
+            ('type', '/type'),
+            ('links', '/links'),
+            ('publisher', '/publisher'),
+            ('publishedDate', '/displayToPublicDate'),
+            ('publicationYear', '/publicationYear'),
+            ('issue', '/issue'),
+            ('volume', '/volume'),
+            ('language', '/language'),
+            ('indexId', '/indexId'),
+            ('publicationSubtype', '/publicationSubtype'),
+            ('startPage', '/startPage'),
+            ('endPage', '/endPage'),
+            ('onlineOnly', '/onlineOnly'),
+            ('additionalOnlineFiles', '/additionalOnlineFiles'),
+            ('country', '/country'),
+            ('state', '/state'),
+            ('ipdsId', '/ipdsId'),
+            ('links', '/links'),
+            ('doi', '/doi'),
+            ('contributors', '/contributors'),
```

Comment:
Actually, this was included here as well because contributors has a whole bunch of info on each author that isn't in the schema, and this seemed the easiest way to capture all of it still attached to each person